concordium-std-derive/src/lib.rs

```diff
@@ -169,13 +401,9 @@ pub fn init(attr: TokenStream, item: TokenStream) -> TokenStream {
 fn init_worker(attr: TokenStream, item: TokenStream) -> syn::Result<TokenStream> {
     let attrs = Punctuated::<Meta, Token![,]>::parse_terminated.parse(attr)?;
 
-    let contract_name = get_attribute_value(attrs.iter(), "contract")?.ok_or_else(|| {
-        syn::Error::new(
-            Span::call_site(),
-            "A name for the contract must be provided, using the contract attribute. For example, \
-             #[init(contract = \"my-contract\")]",
-        )
-    })?;
+    let init_attributes = parse_init_attributes(&attrs)?;
+
+    let contract_name = init_attributes.contract;
 
     let ast: syn::ItemFn =
         attach_error(syn::parse(item), "#[init] can only be applied to functions.")?;
```

Comment:
Consider moving this above the parsing of attributes to avoid unneeded computations and to clear up the error messages.

Reply:
Similar to what is done in `receive_worker`.

Reply:
Done.